### PR TITLE
GCN_48792286

### PR DESCRIPTION
**Pull request by Sylvie1216**
Created by student: 48792286
In this project, my aim is to develop a multi-layer Graph Neural Network (GNN) model to perform semi-supervised multi-class node classification using the Facebook Large Page-Page Network dataset. Achieved test accuracy is 93%.
The directory has the following structure:
-``` README.md ``` -contains how the implemented algorithm works and the problem it solves
-```dataset.py``` -contains the data loader for loading and preprocess my data
-```predict.py``` -provide visualisations where applicable
-```train.py``` -contains the source code for training
-```modules.py```- contains the source code of the components of my model

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme Overall: Good

- Project Overview: Good
- Model Documented/Explained: Good
- Data + Pre-processing: Not done. Used 128 embedding as model input. This is ok.
- Training / Loss Curve(s): Good. Val/Test loss/acc not shown.
- Result Demonstration: Good
- References: Good
- Setup/Reproducibility/Requirements: OK.

Functionality:

- Multi-layer GCN: Good, 2 layers. Uses external code (torch_geometric.nn import GCNConv)
- Semi supervised: Good
- Accuracy: OK. Not mentioned if its Test or Train accuracy.
- TSNE or UMAP: Good.
- Discussion: Good

Code: Has early stop in the code, but also uses multiple epochs as a searchable hyper parameter, which is redundant.

- Consistent with Results: Yes.
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good
- Reusable Modular code/Design: Good

Other Comments:
No early stop.
No hyper parameter search.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Adequate design and implementation. </td><td>-1</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. No early stop.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td>-1</td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>Data leakage found. No data splits.</td><td>-2</td></tr>
	<tr><td></td><td>Difficulty : Normal. GNN Task.</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Some messages are very uninformative (Ex: add details).</td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Usage missing.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Give a  high level intro to the approach instead of the file list.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-15</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Xinyue Yuan
